### PR TITLE
Add finalizers to clean up resources on CR deletion

### PR DESCRIPTION
**Pull request by andykrohg**

### CODE REVIEW

**Comment by andykrohg** [comment]
Requires https://github.com/RedHatGov/devsecops-api-collection/pull/4 to be promoted to `main`

File: playbooks/cleanup-tssc-platform.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+- name: Clean up remaining resources
+  shell: |
+    oc delete -n {{ tssc_namespace }} KeycloakUser,KeycloakClient --all
+    oc delete -n {{ tssc_namespace }} all -l app=jenkins-persistent
```
Comment (jflowers) [inline playbooks/cleanup-tssc-platform.yml:5]: in the past I have found I need to delete jenkins with this command:
oc delete all,configmap,pvc,serviceaccount,rolebinding --selector app=jenkins-persistent -n [...]

File: playbooks/cleanup-tssc-platform.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+- name: Clean up remaining resources
+  shell: |
+    oc delete -n {{ tssc_namespace }} KeycloakUser,KeycloakClient --all
```
Comment (jflowers) [inline playbooks/cleanup-tssc-platform.yml:4]: do we not also need these?:
oc delete KeycloakClient --all -n [...]
oc delete KeycloakRealm --all -n [...]

**Review by jflowers — COMMENTED** [review]
also should we do this?:

oc delete Subscription --all -n [...]
oc delete OperatorGroup --all -n [...]
oc delete ClusterServiceVersion --all -n [...]

File: playbooks/cleanup-tssc-platform.yml (hunk 2 above)
Comment (andykrohg) [reply playbooks/cleanup-tssc-platform.yml:4]: Those are removed in the `rhsso` role when the finalizer passes `cr_state: absent`:
https://github.com/RedHatGov/tssc-operator/blob/2621eae5e9703a815902edd6c4f9843531f6ffec/roles/tssc-platform/rhsso/tasks/rhsso-ocp-idp.yml#L26-L34

File: playbooks/cleanup-tssc-platform.yml (hunk 1 above)
Comment (andykrohg) [reply playbooks/cleanup-tssc-platform.yml:5]: > also should we do this?:
> 
> oc delete Subscription --all -n [...]
> oc delete OperatorGroup --all -n [...]
> oc delete ClusterServiceVersion --all -n [...]

My current approach simply removes CRs, leaving other OLM artifacts in place. Deleting Subscriptions can result in unexpected behavior in the event that the the `tssc-operator` lives in the same namespace as the `TsscPlatform` or `DevSecOpsWorkshop` CR. The operator pod is cleaning up resources when it accidentally deletes itself.

**Review by jflowers — APPROVED** [review]
lgtm

**Pull request merged**